richtextfx/src/main/java/org/fxmisc/richtext/skin/TextExt.java

```diff
@@ -0,0 +1,95 @@
+package org.fxmisc.richtext.skin;
+
+import com.sun.javafx.css.converters.PaintConverter;
+import javafx.beans.property.ObjectProperty;
+import javafx.css.CssMetaData;
+import javafx.css.Styleable;
+import javafx.css.StyleableObjectProperty;
+import javafx.css.StyleableProperty;
+import javafx.scene.paint.Color;
+import javafx.scene.paint.Paint;
+import javafx.scene.text.Text;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TextExt extends Text {
+
+    private ObjectProperty<Paint[]> backgroundColor = null;
+
+    public TextExt(String text) {
+        super(text);
+    }
+
+    @Override
+    public List<CssMetaData<? extends Styleable, ?>> getCssMetaData() {
+        try {
+            // Get inner class
+            Class<?> clazz = Class.forName("javafx.scene.text.Text$StyleableProperties");
+
+            // Get field and make it accessible
+            Field styleablesField = clazz.getDeclaredField("STYLEABLES");
+            styleablesField.setAccessible(true);
+
+            // Get list value and make it modifiable
+            List<CssMetaData<? extends Styleable, ?>> styleables = (List<CssMetaData<? extends Styleable, ?>>) styleablesField.get(null);
```

Comment:
I was trying to manipulate the internal StyleableProperties.STYLEABLES field in a static section before but it didn't work like I imagined. Overriding getCssMetaData was the final solution but I guess I was too lazy to clean the mess up. It's fixed in the latest commit.